### PR TITLE
03.controls added to the example

### PR DESCRIPTION
**Pull request by yescorp**

### CODE REVIEW

**Comment by alexeyinkin** [comment]
`.gitkeep` is never needed for non-empty directories.

File: example/lib/03.control/code_editor.dart [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import 'package:flutter/material.dart';
+
+import 'constants/constants.dart';
+import 'widgets/code_box.dart';
+import 'widgets/code_editor_appbar.dart';
+
+class CodeEditor extends StatefulWidget {
+  final String language;
+  final String theme;
+  final bool reset;
```
Comment (alexeyinkin) [inline example/lib/03.control/code_editor.dart:10]: Android Studio would add a blank line here if you use its formatter. I am not sure if `dart format` adds it or Android Studio, but it's better to add one.

File: example/lib/03.control/code_editor.dart [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  });
+
+  @override
+  State<CodeEditor> createState() => _CodeEditorState();
+}
+
+class _CodeEditorState extends State<CodeEditor> {
+  late String language;
+  late String theme;
+  late bool reset;
```
Comment (alexeyinkin) [inline example/lib/03.control/code_editor.dart:25]: Unused?

File: example/lib/03.control/code_editor.dart [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    required this.reset,
+  });
+
+  @override
+  State<CodeEditor> createState() => _CodeEditorState();
+}
+
+class _CodeEditorState extends State<CodeEditor> {
+  late String language;
+  late String theme;
```
Comment (alexeyinkin) [inline example/lib/03.control/code_editor.dart:24]: https://medium.com/flutter-senior/avoiding-late-variables-in-dart-abdb4c2b57e7

Just initialize them with lists' first values.

File: example/lib/03.control/code_editor.dart [hunk 4]
```diff
@@ -0,0 +1,77 @@
+import 'package:flutter/material.dart';
+
+import 'constants/constants.dart';
+import 'widgets/code_box.dart';
+import 'widgets/code_editor_appbar.dart';
+
+class CodeEditor extends StatefulWidget {
+  final String language;
+  final String theme;
```
Comment (alexeyinkin) [inline example/lib/03.control/code_editor.dart:9]: This blurs sources of truth for these values.

File: example/lib/03.control/code_editor.dart [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          shouldContainInitialText: true,
+          theme: theme,
+        ),
+      ),
+    );
+  }
+
+  void _onLanguageChanged(String? value) {
+    setState(() {
+      language = value ?? language;
```
Comment (alexeyinkin) [inline example/lib/03.control/code_editor.dart:61]:
```suggestion
  void _onLanguageChanged(String value) {
    setState(() {
      language = value;
```

File: example/lib/03.control/code_editor.dart [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    });
```
Comment (alexeyinkin) [inline example/lib/03.control/code_editor.dart:62]: What will change the language in the controller?

File: example/lib/03.control/main.dart [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import 'constants/constants.dart';
+
+void main() {
+  runApp(const MyApp());
+}
+
+final controller = CodeController(
+  text: javaFactorialSnippet,
+  language: java,
+);
```
Comment (alexeyinkin) [inline example/lib/03.control/main.dart:21]: Unused?

File: example/lib/03.control/widgets/code_box.dart [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+import 'package:flutter_code_editor/flutter_code_editor.dart';
+
+import '../../common/snippets.dart';
+import '../../common/themes.dart';
+import '../constants/constants.dart';
+
+class CodeBox extends StatefulWidget {
+  final String language;
+  final String theme;
+  final bool shouldContainInitialText;
```
Comment (alexeyinkin) [inline example/lib/03.control/widgets/code_box.dart:11]: Examples should be short. This is not required to showcase the switching.

File: example/lib/03.control/widgets/code_editor_appbar.dart [hunk 9]
```diff
@@ -0,0 +1,70 @@
+import 'package:flutter/material.dart';
+
+import 'dropdown_selector.dart';
+
+class CodeEditorAppbar extends StatelessWidget implements PreferredSizeWidget {
+  final double height;
+  final List<String?> languages;
+  final List<String?> themes;
+  final String? title;
+  final Function(String?)? onLanguageChanged;
```
Comment (alexeyinkin) [inline example/lib/03.control/widgets/code_editor_appbar.dart:10]:
```suggestion
  final ValueChanged<String> onLanguageChanged;
```

File: example/lib/03.control/widgets/code_editor_appbar.dart [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    super.key,
+    required this.height,
+    required this.languages,
+    required this.themes,
+    required this.selectedLanguage,
+    required this.selectedTheme,
+    this.title,
+    this.onLanguageChanged,
+    this.onThemeChanged,
+    this.onReset,
```
Comment (alexeyinkin) [inline example/lib/03.control/widgets/code_editor_appbar.dart:28]: https://medium.com/flutter-senior/sort-alphabetically-everything-you-can-1e9d7299c4ae

Check everything else to sort in the app.

File: example/lib/03.control/widgets/code_editor_appbar.dart [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      height: MediaQuery.of(context).size.height / 13,
```
Comment (alexeyinkin) [inline example/lib/03.control/widgets/code_editor_appbar.dart:34]: Can we go with a fixed height?

File: example/lib/03.control/widgets/dropdown_selector.dart [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return DropdownButton<String>(
+      value: value,
+      items: choices.map((String? value) {
+        return DropdownMenuItem<String>(
+          value: value,
+          child: value == null
+              ? const Divider()
```
Comment (alexeyinkin) [inline example/lib/03.control/widgets/dropdown_selector.dart:25]: Does this mean the currently selected item is not visible in the dropdown?

File: example/lib/03.control/widgets/dropdown_selector.dart [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+              : Text(value, style: const TextStyle(color: Colors.white)),
+        );
+      }).toList(),
```
Comment (alexeyinkin) [inline example/lib/03.control/widgets/dropdown_selector.dart:28]: https://medium.com/flutter-senior/always-use-non-growable-arrays-if-possible-4864a022a54a

**Review by alexeyinkin — CHANGES_REQUESTED** [review]

File: example/lib/03.control/widgets/dropdown_selector.dart (hunk 12 above)
Comment (yescorp) [reply example/lib/03.control/widgets/dropdown_selector.dart:25]: No, seems like it doesn't do anything
Removed it.

File: example/lib/03.control/code_editor.dart (hunk 5 above)
Comment (yescorp) [reply example/lib/03.control/code_editor.dart:56]: I need to pass Function(String?) as parameter. Is it possible to do this somehow?

File: example/lib/03.control/widgets/code_editor_appbar.dart (hunk 9 above)
Comment (yescorp) [reply example/lib/03.control/widgets/code_editor_appbar.dart]: I need `Function(String?)` so changed to `ValueChanged<String?>`

File: example/lib/03.control/code_editor.dart (hunk 5 above)
Comment (alexeyinkin) [reply example/lib/03.control/code_editor.dart:56]:
```dart
onChanged: (value) { if (value != null) onChanged(value); },
```

**Pull request closed without merging**